src/sickle.ts

```diff
@@ -66,6 +77,34 @@ class Annotator {
           this.writeNode(node);
         }
         break;
+      case ts.SyntaxKind.InterfaceDeclaration:
+        this.writeRange(node.getFullStart(), node.getEnd());
+        if (node.flags & ts.NodeFlags.Ambient) {
+          // It's a "declare interface ..."; write a Closure interface
+          // declaration so we can @export all properties.
+          let decl = <ts.InterfaceDeclaration>node;
+          let className = decl.name.getText();
+          let props: TypedProperty[] = [];
+          for (let member of decl.members) {
+            // Members can include things like index signatures, for e.g.
+            //   interface Foo { [key: string]: number; }
+            // For now, just die unless all the members are regular old
+            // properties.
+            if (member.kind != ts.SyntaxKind.PropertySignature) {
+              this.fail(`unhandled member of kind ${ts.SyntaxKind[member.kind]}`);
+            }
+            let prop = <ts.PropertySignature>member;
+            props.push({
+              name: prop.name.getText(),
+              type: prop.type,
+              extraTags: '@export',
+            });
+          }
+          this.emit(`\nclass ${className} {\n`);
```

Comment:
A bit more on the class thing: when I use a class, Closure figures out all the mapping of "class Foo" to its @interface/@struct/etc stuff.  And I can just emit the ES6-style annotations helper as a static method.  If I don't emit a class, I have to create an interface type and I have to figure out where to stuff the declarations of the fields.  (I guess I put them in the ctor?  I will try it.)